core/templates/pages/classroom-admin-page/modals/create-new-classroom-modal.component.ts

```diff
@@ -0,0 +1,174 @@
+
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Create new classroom modal.
+ */
+
+import { Component } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { AppConstants } from 'app.constants';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { ClassroomBackendApiService } from '../../../domain/classroom/classroom-backend-api.service';
+
+
+@Component({
+  selector: 'oppia-create-new-classroom-modal',
+  templateUrl: './create-new-classroom-modal.component.html'
+})
+export class CreateNewClassroomModalComponent
+  extends ConfirmOrCancelModal {
+  constructor(
+    private classroomBackendApiService: ClassroomBackendApiService,
+    private ngbActiveModal: NgbActiveModal
+  ) {
+    super(ngbActiveModal);
+  }
+
+  existingClassroomNames: string[] = [];
+  newClassroomName: string = '';
+  newClassroomId: string = '';
+  newClassroomUrlFragment: string = '';
+  creatingNewClassroom: boolean = false;
+
+  classroomNameExceedsMaxLen: boolean = false;
+  emptyClassroomName: boolean = false;
+  duplicateClassroomName: boolean = false;
+  classroomNameIsValid: boolean = false;
+
+  classroomUrlFragmentExceedsmaxLen: boolean = false;
+  emptyClassroomUrlFrgament: boolean = false;
+  duplicateClassroomUrlFragment: boolean = false;
+  urlFragmentRegexMatched: boolean = true;
+  classroomUrlFragmentIsValid: boolean = false;
+
+  getNewClassroomId(): void {
+    this.classroomBackendApiService.getNewClassroomIdAsync().then(
+      classroomId => {
+        this.newClassroomId = classroomId;
+      }
+    );
+  }
+
+  ngOnInit(): void {
+    this.getNewClassroomId();
```

Comment:
Should we only allow user interaction with the component once the classroom ID is set?

Reply:
Yes, the interaction should be performed after fetching the classrooms from the backend.

Reply:
Thank you for the explanation, I have updated the code, please take a look.